app/auth/security.py

```diff
@@ -0,0 +1,93 @@
+from typing import Union, Annotated
+from datetime import timezone, timedelta, datetime
+
+from passlib.context import CryptContext
+import jwt
+from jwt.exceptions import InvalidTokenError
+from fastapi.security import OAuth2PasswordBearer
+from fastapi import HTTPException, Depends, status
+from ..models.users import Users, TokenData
+from app.db import engine
+from sqlmodel import Session
+from app.config import Config
+
+ALGORITHM = "HS256"
+ACCESS_TOKEN_EXPIRE_MINUTES = 30
+
+pwd_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
+
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")
+
+def verify_password(plain_password, hashed_password):
+    return pwd_context.verify(plain_password, hashed_password)
+
+def get_password_hash(password):
+    return pwd_context.hash(password)
+
+def get_user(username: str):
+    with Session(engine) as session:
+        
+        user = session.get(Users, username)
+        
+        return user
+
+def authenticate_user(username: str, password: str):
```

Comment:
Can this read `def authenticate_user(username: str, password: str) -> User | bool` to indicate that this method can return both these data types.